### ISSUE TITLE
Any way to decompose OptionParse.opt blocks?

### ISSUE THREAD

**Issue by williamchan** [scopt/scopt#54]
Thanks for sharing this repo! I have a question. I have code that looks like this:

``` scala
val parser = new OptionParser[MyConfig](...) {
  ...
    cmd("command0") action { (_, c) => c.copy(command = "command0") } text "..." children {
      opt[Option[String]]('p', "my-parameter") { ... }
    }
    cmd("command1") action { (_, c) => c.copy(command = "command1") } text "..." children {
      opt[Option[String]]('p', "my-parameter") { ... }
    }
}
```

Notice the code repeated between the two cmds. I would like to be able to refactor out the "opt[Option[String]]('p',) { ... }" so as to not repeat myself, but I have been having trouble figuring that out. Is there a way to refactor this code?

**Comment by eed3si9n** [comment]
It's not currently possible, but I might try some ideas to make that happen hopefully without breaking existing code too much.

**Comment by eed3si9n** [comment]
Released scopt 4.0.0-RC2 with more reusable OParser. See http://eed3si9n.com/scopt4 for details.
